### PR TITLE
fix(python): deprecate `time_unit` property from `Series`

### PR DESCRIPTION
**Pull request by tim-habitat**
closes https://github.com/pola-rs/polars/issues/8989

### CODE REVIEW

File: py-polars/polars/series/series.py [hunk 1]
```diff
@@ -403,6 +403,11 @@ def shape(self) -> tuple[int]:
     @property
     def time_unit(self) -> TimeUnit | None:
         """Get the time unit of underlying Datetime Series as {"ns", "us", "ms"}."""
+        warnings.warn(
+            "In a future version of polars, the `time_unit` attribute will be removed",
```
Comment (MarcoGorelli) [inline py-polars/polars/series/series.py:407]: could you advise what to use instead too please?

Comment (tim-habitat) [reply py-polars/polars/series/series.py]: sorry for taking up your time, realise that you may have done it faster than just the review 😅
I'll convert to draft until i fix the tests too - i struggled to set up the environment but i think i'm gonna figure it out now

File: py-polars/pyproject.toml [hunk 2]
```diff
@@ -184,6 +184,8 @@ filterwarnings = [
   "error",
   # ...except where it prevents test debugging in an IPython console
   "ignore:.*unrecognized arguments.*PyDevIPCompleter:DeprecationWarning",
+  # or for this deprecation warning
+  "ignore:`Series.time_unit` is deprecated and will be removed in a future version, please use `Series.dtype.time_unit` instead:DeprecationWarning",
```
Comment (tim-habitat) [inline py-polars/pyproject.toml:188]: Without this ignore, 42 tests fails. I'm not sure whether we should put `warnings` catch on all of them or just ignore them from here is fine ?

File: py-polars/polars/series/series.py [hunk 3]
```diff
@@ -403,6 +403,12 @@ def shape(self) -> tuple[int]:
     @property
     def time_unit(self) -> TimeUnit | None:
         """Get the time unit of underlying Datetime Series as {"ns", "us", "ms"}."""
+        warnings.warn(
+            "`Series.time_unit` is deprecated and will be removed in a future version,"
+            " please use `Series.dtype.time_unit` instead",
+            category=DeprecationWarning,
+            stacklevel=2,
```
Comment (MarcoGorelli) [inline py-polars/polars/series/series.py:410]: can you use `find_stacklevel`

File: py-polars/pyproject.toml (hunk 2 above)
Comment (MarcoGorelli) [reply py-polars/pyproject.toml:188]: which ones fail? I suspect it's cases when internally polars is using `.time_unit` - in which case, you might be able to update that to `.dtype.time_unit` and avoid the warning

Comment (MarcoGorelli) [reply py-polars/pyproject.toml:188]: yeah I think type: ignore is fine, as not all dtypes have `time_unit` and `time_zone`

**Comment by MarcoGorelli** [comment]
also, you'll need to run `make pre-commit` from the `py-polars` folder

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
almost there 💪

File: py-polars/pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tim-habitat) [reply py-polars/pyproject.toml]: sorry forgot to commit this one

File: py-polars/tests/unit/datatypes/test_temporal.py [hunk 4]
```diff
@@ -486,7 +486,8 @@ def test_date_range() -> None:
             time_unit=time_unit,
             eager=True,
         )
-        assert rng.time_unit == time_unit
+        with pytest.warns(DeprecationWarning):
```
[the remark this answers is not in the document]
Comment (tim-habitat) [reply py-polars/tests/unit/datatypes/test_temporal.py]: great idea, done that

**Review by MarcoGorelli — APPROVED** [review]
Nice one, thanks @tim-habitat !

**Comment by ritchie46** [comment]
Thanks @tim-habitat  and thank for the review @MarcoGorelli

**Comment by tim-habitat** [comment]
Thanks both for the great library and your patience on this first contribution 🙏

**Pull request merged**